Results_Engine/Query/DisplayMeshResults.cs

```diff
@@ -0,0 +1,214 @@
+                            Face newFace = new Face()
+                            {
+                                A = faceReference.A == k ? verts.Count - 1 : faceResult.A,
+                                B = faceReference.B == k ? verts.Count - 1 : faceResult.B,
+                                C = faceReference.C == k ? verts.Count - 1 : faceResult.C,
+                                D = faceReference.IsQuad() ? faceReference.D == k ? verts.Count - 1 : faceResult.D : -1
+                            };
+                            faceDictionaryResult[FaceRelatedValue.Key] = newFace;
+                        }
+                    }
+                    faces = faceDictionaryResult.Select(x => x.Value).ToList();
+                    break;
+
+                case MeshResultSmoothingType.ByPanel:
+                case MeshResultSmoothingType.Global:
+                    List<double> nodeValues = tempMappedElementResults.Select(x => x[0].ResultToValue(meshResultDisplay)).ToList();
+
+                    // Add all verticies to a list with their colour
+                    for (int k = 0; k < mesh.Nodes.Count; k++)
+                    {
+                        verts.Add(new Vertex()
+                        {
+                            Point = mesh.Nodes[k].Position,
+                            Color = gradient.Color(nodeValues[k], from, to)
+                        });
+                    }
+                    faces = mesh.Faces.Geometry().ToList();
+                    break;
+
+                case MeshResultSmoothingType.ByFiniteElementCentres:
+                case MeshResultSmoothingType.BySelection:
+                default:
+                    Engine.Reflection.Compute.RecordError("Unsupported SmoothingType: " + Reflection.Query.PropertyValue(meshResult, "Smoothing").ToString() +
+                                                      " detected, meshResult for ObjectId: " + meshResult.ObjectId.ToString() +
+                                                      " and ResultCase: " + meshResult.ResultCase.ToString() + "will be returned empty.");
+                    return new RenderMesh();
+            }
+
+            return new RenderMesh() { Vertices = verts, Faces = faces };
+        }
+
+        /***************************************************/
+
+        /*
+        private static Func<TMeshElementResult, double> GetPropertyDelegate<TMeshElementResult>(TMeshElementResult meshResult, string meshResultDisplay) where TMeshElementResult : IMeshElementResult
+        {
+            PropertyInfo prop = meshResult.GetType().GetProperty(meshResultDisplay);
+
+            if (prop == null)
+                return null;
+            return Delegate.CreateDelegate(typeof(Func<TMeshElementResult, double>), null, prop.GetGetMethod()) as Func<TMeshElementResult, double>;
+        }*/
+
+        /***************************************************/
+
```

Comment:
Any good reason to keep this commented out private method? Or should it be cleaned out?

Reply:
This is @IsakNaslundBh who attempted to create a more optimised replacement for my calls to PropertyValue, but we never got it to work and when I read up about it later I got the impression that it wouldn't work in this case, though it was a while ago and I don't remember the details. Personally I haven't noticed the PropertyValue calls slowing this method down one bit, so I'd be happy to remove it.

Reply:
I think you can delete this. If the performance becomes a problem, a separate PR will target this.